test/e2e/operator_test.go

```diff
@@ -158,11 +107,316 @@ func ensureCredentialsRequest() error {
+					t.Fatalf("failed to close reader for pod %s: %v", clientPod.Name, err)
+				}
+			}()
+			for scanner.Scan() {
+				line := scanner.Text()
+				if strings.Contains(line, "HTTP/1.1 200 OK") {
+					return true, nil
+				}
+			}
+			return false, nil
+		})
+		if err != nil {
+			t.Fatalf("failed to observe the expected log message: %v", err)
+		}
+		defer func() {
+			err = kubeClient.Delete(context.TODO(), clientPod, &client.DeleteOptions{PropagationPolicy: &deletetionPolicy})
+			if err != nil {
+				t.Fatalf("failed to delete clientpod %s: %v", clientPod.Name, err)
+			}
+			t.Logf("deleted echo clientpod %s", clientPod.Name)
+		}()
+	}
+}
+
+func TestAWSLoadBalancerControllerWithCustomIngressClass(t *testing.T) {
+	t.Log("Verifying operator availability due to restarts caused by deletion of resources")
+	getOperator(t)
+
+	t.Log("Creating a custom ingress class")
+	ingclassName := types.NamespacedName{Name: "custom-alb", Namespace: "aws-load-balancer-operator"}
+	ingclass := buildIngressClass(ingclassName, "ingress.k8s.aws/alb")
+	if err := kubeClient.Create(context.TODO(), ingclass); err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("failed to ensure custom ingress class %q: %v", ingclass.Name, err)
+	}
+	defer func() {
+		err := kubeClient.Delete(context.TODO(), ingclass, &client.DeleteOptions{PropagationPolicy: &deletetionPolicy})
+		if err != nil {
+			t.Fatalf("failed to delete custom ingress class %q: %v", ingclass.Name, err)
+		}
+		t.Logf("deleted custom ingress class: %s", ingclass.Name)
+	}()
+
+	t.Log("Creating aws load balancer controller instance with custom ingress class")
+
+	name := types.NamespacedName{Name: "cluster", Namespace: "aws-load-balancer-operator"}
+	alb := newAWSLoadBalancerController(name, ingclassName.Name, []albo.AWSAddon{})
+	if err := kubeClient.Create(context.TODO(), &alb); err != nil {
+		if errors.IsAlreadyExists(err) {
+			currentAlb := albo.AWSLoadBalancerController{}
+			if err := kubeClient.Get(context.TODO(), name, &currentAlb); err != nil {
+				t.Fatalf("failed to get aws load balancer controller %q: %v", name, err)
+			}
+			currentAlb.Spec.IngressClass = ingclassName.Name
+			if err = kubeClient.Update(context.TODO(), &currentAlb); err != nil {
+				t.Fatalf("failed to update aws load balancer controller %q: %v", name, err)
+			}
+		} else {
+			t.Fatalf("failed to create aws load balancer controller %q: %v", name, err)
+		}
+	}
+
 	expected := []appsv1.DeploymentCondition{
 		{Type: appsv1.DeploymentAvailable, Status: corev1.ConditionTrue},
 	}
-	if err := waitForOperatorDeploymentStatusCondition(t, kubeClient, expected...); err != nil {
-		t.Errorf("Did not get expected available condition: %v", err)
+	deploymentName := types.NamespacedName{Name: "aws-load-balancer-controller-cluster", Namespace: "aws-load-balancer-operator"}
+	if err := waitForDeploymentStatusConditionUntilN(t, kubeClient, time.Minute*30, 20, deploymentName, expected...); err != nil {
```

Comment:
Does it really have to be 30 minutes? 5 minutes seems reasonable. Setting it to 30 just makes a failing test take longer to fail.